test/infoblox_provider_test.rb

```diff
@@ -18,6 +18,23 @@ def setup
 
     @network_2 = Infoblox::Network.new(:network => '192.168.43.0/24')
     @provider = Proxy::DHCP::Infoblox::Provider.new(@connection, @crud, @restart_grid, @unused_ips, @managed_subnets)
+
+    @lease = Infoblox::Lease.new(:address => '192.168.43.127',
+                                 :binding_state => 'active',
+                                 :client_hostname => 'example',
+                                 :hardware => 'ba:be:fa:ce:ca:fe',
+                                 :network => @subnet.network,
+                                 :starts => nil,
+                                 :ends => nil)
+  end
+
+  def test_leases
+    @crud = ::Proxy::DHCP::Infoblox::CommonCRUD.new(@connection)
+    @provider = Proxy::DHCP::Infoblox::Provider.new(@connection, @crud, @restart_grid, @unused_ips, @managed_subnets)
+
+    @provider.expects(:full_network_address).with(@subnet.network).returns(@network.network)
+    Infoblox::Lease.expects(:find).with(@connection, 'address~' => "192\\.168\\.42\\..+").returns([@lease])
+    assert_equal @lease.hardware, @provider.all_leases(@subnet.network).first.mac
   end
```

Comment:
Can you also add a test for a subnet that is not class A-B-C? Like netmask `255.255.200.0` or something?